### PR TITLE
[WIP] WD-4072 Automate Badge Issuing using Credly

### PR DESCRIPTION
**Pull request by abhigyanghosh30**
## Done

- [List of work items including drive-bys - remember to add the why and what of this work.]

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/
    - Be sure to test on mobile, tablet and desktop screen sizes
### Proper QA
- Take the sample exam on TA 
    - Pass the said exam
    - After a few hours, a badge should appear in /credentials/your-badges page
### Easier QA
- Visit https://app.trueability.com/m/ability_screens/4190/webhooks/202/test
    - Response should be {"status":"badge_not_issued"}, 403
    - This means that the website is accepting the webhooks
- Now call the [same endpoint](https://ubuntu-com-12946.demos.haus/credentials/assessment_passed) using cURL/Postman.  
    `{
    "assessment":{
        "score":0.75,
        "user":{
            "full_name":"John Doe",
            "email":"john.doe@xyz.com"
        },
        "ability_screen":{
            "cutoff_score": 0.6
        },
        "ability_screen_variant":{
            "ability_screen_id":4190
        }
    }
}`  
    - A badge should be issued from Credly
    - Verify new badge in [/your-badges](https://ubuntu-com-12946.demos.haus/credentials/your-badges)

## Issue / Card

Fixes [WD-4072](https://warthogs.atlassian.net/browse/WD-4072)

## Screenshots

[If relevant, please include a screenshot.]

## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)

[WD-4072]: https://warthogs.atlassian.net/browse/WD-4072?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mrgnr** [comment]
@abhigyanghosh30 The demo is giving a [500 error](https://sentry.is.canonical.com/canonical/ubuntu-com/issues/34348/events/2329657/) on https://ubuntu-com-12946.demos.haus/credentials/your-badges -- I think we need to add `CREDLY_TOKEN` to the [konf](https://github.com/abhigyanghosh30/ubuntu.com/blob/trueability-credly-integration/konf/site.yaml).

I passed an exam and am able to test locally, but just waiting for the badge to appear.

**Comment by abhigyanghosh30** [comment]
Yes. That is an issue. Do you know how to do that? I don't have access to it

**Comment by mrgnr** [comment]
Someone with K8S access (@jpmartinspt or @albertkol ?) will need to add the secret to the demo cluster and konf/site.yaml will need to be updated.

**Comment by jpmartinspt** [comment]
I can set that up.

File: webapp/shop/cred/views.py [hunk 1]
```diff
@@ -678,3 +679,103 @@ def cred_beta_activation(**_):
...
+
+
+@shop_decorator(area="cred", permission="guest", response="json")
+def issue_badges(trueability_api, credly_api, **kwargs):
+    webhook_response = flask.request.json
+    api_key = flask.request.headers.get("X-API-KEY")
+    if not api_key or api_key != os.getenv("TA_WEBHOOK_API_KEY"):
+        return flask.jsonify({"status": "Invalid API Key"}), 401
+    assessment_score = webhook_response["assessment"]["score"]
+    print(assessment_score)
```
Comment (jpmartinspt) [inline webapp/shop/cred/views.py:752]: Remove print here.

File: webapp/shop/cred/views.py [hunk 2]
```diff
@@ -678,3 +679,107 @@ def cred_beta_activation(**_):
...
+
+
+@shop_decorator(area="cred", permission="guest", response="json")
+def issue_badges(trueability_api, credly_api, **kwargs):
+    webhook_response = flask.request.json
+    api_key = flask.request.headers.get("X-API-KEY")
+    if not api_key or api_key != os.getenv("TA_WEBHOOK_API_KEY"):
+        return flask.jsonify({"status": "Invalid API Key"}), 401
+    assessment_score = webhook_response["assessment"]["score"]
+    cutoff_score = webhook_response["ability_screen"]["cutoff_score"]
```
Comment (mrgnr) [inline webapp/shop/cred/views.py:752]: I had to change this line to the following when testing locally:

```
cutoff_score = webhook_response["assessment"]["ability_screen"]["cutoff_score"]
```

**Comment by mrgnr** [comment]
I QA'd locally using the "Easier QA" since the demo isn't working. I had to make one change to the code, then it worked nicely.

**Pull request merged**